libs/langchain/langchain/memory/chat_message_histories/singlestoredb.py

```diff
@@ -0,0 +1,263 @@
+                    SingleStoreDBChatMessageHistory
+                )
+
+                message_history = SingleStoreDBChatMessageHistory(
+                    session_id="my-session",
+                    host="127.0.0.1",
+                    port=3306,
+                    user="user",
+                    password="password",
+                    database="db",
+                    table_name="my_custom_table",
+                    pool_size=10,
+                    timeout=60,
+                )
+
+            Using environment variables:
+
+            .. code-block:: python
+
+                from langchain.memory.chat_message_histories import (
+                    SingleStoreDBChatMessageHistory
+                )
+
+                os.environ['SINGLESTOREDB_URL'] = 'me:p455w0rd@s2-host.com/my_db'
+                message_history = SingleStoreDBChatMessageHistory("my-session")
+        """
+
+        self.table_name = table_name
+        self.session_id = session_id
+        self.id_field = id_field
+        self.session_id_field = session_id_field
+        self.message_field = message_field
+
+        """Pass the rest of the kwargs to the connection."""
+        self.connection_kwargs = kwargs
+
+        """Add connection attributes to the connection kwargs."""
+        if "conn_attrs" not in self.connection_kwargs:
+            self.connection_kwargs["conn_attrs"] = dict()
+
+        self.connection_kwargs["conn_attrs"]["_connector_name"] = "langchain python sdk"
+        self.connection_kwargs["conn_attrs"]["_connector_version"] = "1.0.1"
+
+        """Create a connection pool."""
+        try:
+            from sqlalchemy.pool import QueuePool
+        except ImportError:
+            raise ImportError(
+                "Could not import sqlalchemy.pool python package. "
+                "Please install it with `pip install singlestoredb`."
+            )
+
+        self.connection_pool = QueuePool(
+            self._get_connection,
+            max_overflow=max_overflow,
+            pool_size=pool_size,
+            timeout=timeout,
+        )
+        self._create_table_if_not_exists()
+
+    def _get_connection(self) -> Any:
+        try:
+            import singlestoredb as s2
+        except ImportError:
+            raise ImportError(
+                "Could not import singlestoredb python package. "
+                "Please install it with `pip install singlestoredb`."
+            )
+        return s2.connect(**self.connection_kwargs)
+
+    def _create_table_if_not_exists(self) -> None:
+        """Create table if it doesn't exist."""
+        conn = self.connection_pool.connect()
+        try:
+            cur = conn.cursor()
+            try:
+                cur.execute(
+                    """CREATE TABLE IF NOT EXISTS {}
+                    ({} BIGINT PRIMARY KEY AUTO_INCREMENT,
+                    {} TEXT NOT NULL,
+                    {} JSON NOT NULL);""".format(
```

Comment:
Could you add sanitization here for table_name / id field / session_id_field and message field? Assume init parameters cannot be trusted as we do not know how end users intend use the code (e.g., the init parameters may be loaded from somewhere or received from another entity)

Reply:
I agree, we cannot assume these parameters are safe. I added `_sanitize_input` for each parameter in the constructor.